### PR TITLE
Add msgpack serialization

### PR DESCRIPTION
**Pull request by pfultz2**

### CODE REVIEW

**Comment by pfultz2** [comment]
The code coverages fails because of the `debug_print`. I dont think a test case is really needed for that.

File: src/msgpack.cpp [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+            {
+                v = o.as<double>();
+                break;
+            }
+            case msgpack::type::STR:
+            {
+                v = o.as<std::string>();
+                break;
+            }
+            case msgpack::type::BIN: { throw msgpack::type_error();
```
Comment (scxiao) [inline src/msgpack.cpp:48]: Which error message is printed here? can we use the MIGRAPHX_THROW instead?

File: src/msgpack.cpp [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+                migraphx::value r = migraphx::value::object{};
+                std::for_each(o.via.map.ptr,
+                              o.via.map.ptr + o.via.map.size,
+                              [&](const msgpack::object_kv& p) {
+                                  r[p.key.as<std::string>()] = p.val.as<migraphx::value>();
+                              });
+                v = r;
+                break;
+            }
+            case msgpack::type::EXT: { throw msgpack::type_error();
```
Comment (scxiao) [inline src/msgpack.cpp:71]: Same as above

File: src/msgpack.cpp [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+                migraphx::value r = migraphx::value::object{};
+                std::for_each(o.via.map.ptr,
+                              o.via.map.ptr + o.via.map.size,
+                              [&](const msgpack::object_kv& p) {
+                                  r[p.key.as<std::string>()] = p.val.as<migraphx::value>();
+                              });
+                v = r;
+                break;
+            }
+            case msgpack::type::EXT: { MIGRAPHX_THROW("msgpack BIN type not supported.");
```
Comment (scxiao) [inline src/msgpack.cpp:71]: I think the msssage should be "msgpack EXT type not support", not BIN type

Comment (pfultz2) [reply src/msgpack.cpp:71]: Ah good catch!

**Pull request merged**